core/server/controllers/admin.js

```diff
@@ -32,27 +30,7 @@ adminControllers = {
             });
         }
 
-        updateCheck().then(function then() {
-            return updateCheck.showUpdateNotification();
-        }).then(function then(updateVersion) {
-            if (!updateVersion) {
-                return;
-            }
-
-            var notification = {
-                type: 'upgrade',
-                location: 'settings-about-upgrade',
-                dismissible: false,
-                status: 'alert',
-                message: i18n.t('notices.controllers.newVersionAvailable',
-                                {version: updateVersion, link: '<a href="http://support.ghost.org/how-to-upgrade/" target="_blank">Click here</a>'})};
-
-            return api.notifications.browse({context: {internal: true}}).then(function then(results) {
-                if (!_.some(results.notifications, {message: notification.message})) {
-                    return api.notifications.add({notifications: [notification]}, {context: {internal: true}});
-                }
-            });
-        }).finally(function noMatterWhat() {
+        updateCheck().finally(function noMatterWhat() {
```

Comment:
I agree, there's a lot of room for improvement here. I'd refactor the whole update checking logic and design it differently. And sending a request in the background is the way it's supposed to be! Also most of this can be done in ghost-admin.

However, note that the intention behind this PR is just to remove notifications user has already seen. Also, I tried to keep the update check logic in one place as opposed to it being spread between a controller and a util function so it kinda reads like a sequence of steps. The old code  was a bit hard for me follow what was going on. The logic itself hasn't changed though.

Perhaps we can collect these ideas you guys suggested in a ticket and implement them with the goal of improving the update check logic?